### DISCUSSION TITLE
How slugify cyrillic to latin in remark TOC?

### DISCUSSION THREAD

**Discussion by AliIslamov** [remarkjs/remark#1292] [Q&A]
By default internal links stay in cyrillic.

**Comment by ChristianMurphy** [comment]
Welcome @AliIslamov!

Non-latin characters are valid in slugs and are used by Github itself.
Remark TOC intentionally follows the GFM standard, there are no plans to make thar configurable.

As a whole this feels like an XY problem.
https://xyproblem.info/
Why do you want characters converted to a different character set?
What is the problem you are trying to solve?

**Reply by AliIslamov** [reply]
> What is the problem you are trying to solve?

If I sharing internal cyrillic link I get something like this:

https://sahih.faith/posts/argumentaciya-v-islame/#%D1%81%D1%82%D1%80%D1%83%D0%BA%D1%82%D1%83%D1%80%D0%B0-%D0%BE%D1%82%D0%BA%D1%80%D0%BE%D0%B2%D0%B5%D0%BD%D0%B8%D1%8F

I think that it is not good, because such links looks like terrible and uninformative.

For example the same link in latin:

https://sahih.faith/posts/argumentaciya-v-islame/struktura-otkroveniya

**Reply by wooorm** [reply]
The URL you posted is <https://sahih.faith/posts/argumentaciya-v-islame/#структура-откровения>, which works, and this tool doesn’t change that, and it’s what you want. Hence the XY problem Christian mentions.

You can share encoded URLs if you want. But you can also choose to share URLs with non-ascii characters in them. Your choice.

**Reply by AliIslamov** [reply]
> You can share encoded URLs if you want. But you can also choose to share URLs with non-ascii characters in them. Your choice.

If you try to pickup the url in browser, you always pickup it in encoded way by default.

So, to share it in non encoded ciryllic way, I must write it by hands.

Ok, thanks anyway ans sorry for XY.

I will fix this problem myself.

**Reply by wooorm** [reply]
That's not what happens in my browser! I pasted your encoded url and copied again, then it was just Cyrillic

**Reply by AliIslamov** [reply]
> That's not what happens in my browser! I pasted your encoded url and copied again, then it was just Cyrillic

Very interesting.

What browser are you using?

I do test on mobile Chrome, because it generates more than 2/3 of all traffic.

**Reply by ChristianMurphy** [reply]
On FireFox 123 it works as @wooorm  without any tweaking
Chrome 122 takes some tweaking https://stackoverflow.com/questions/18176661/copying-a-utf-8-url-from-browsers-address-bar-gives-only-the-ugly-encoded-one

In both cases the encoded and decoded version of the URL are both valid, the stylistic choice of which to include is yours.
